bundles/CoreBundle/src/DependencyInjection/Configuration.php

```diff
@@ -153,20 +153,36 @@ private function addMaintenanceNode(ArrayNodeDefinition $rootNode): void
     {
         $rootNode
             ->children()
-            ->arrayNode('maintenance')
-            ->addDefaultsIfNotSet()
-            ->children()
-                ->arrayNode('housekeeping')
+                ->arrayNode('maintenance')
                 ->addDefaultsIfNotSet()
-                ->children()
-                    ->integerNode('cleanup_tmp_files_atime_older_than')
-                        ->info('Integer value in seconds.')
-                        ->defaultValue(7_776_000) // 90 days
-                    ->end()
-                    ->integerNode('cleanup_profiler_files_atime_older_than')
-                        ->info('Integer value in seconds.')
-                        ->defaultValue(1800)
+                    ->children()
+                        ->arrayNode('housekeeping')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                                ->integerNode('cleanup_profiler_files_atime_older_than')
+                                    ->info('Integer value in seconds.')
+                                    ->defaultValue(1800)
+                                ->end()
+                            ->end()
+                        ->end()
+                        ->arrayNode('cleanup_directory')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                                ->integerNode('cleanup_files_atime_older_than')
+                                    ->info('Integer value in seconds.')
+                                    ->defaultValue(7_776_000) // 90 days
+                                ->end()
+                                ->arrayNode('directories')
+                                    ->info('Directories to cleanup.')
+                                    ->scalarPrototype()->end()
+                                    ->defaultValue([PIMCORE_PRIVATE_VAR. '/tmp', PIMCORE_WEB_ROOT . '/var/tmp'])
```

Comment:
I don't think clearing the full `public/var/tmp` folder is recomended. 
From [Pimcore documentation](https://pimcore.com/docs/platform/Pimcore/Administration_of_Pimcore/Cleanup_Data_Storage/#temporary-files):

>  WARNING: Deleting all files in public/var/tmp/ can have a huge impact on performance until all needed thumbnails are generated again.

Probably we need something more sophisticated for this folder, e.g. looping over all files and verifying the asset and or thumbnail config still exists

Reply:
well yes it is not recommend to delete it just right away but in this case it would be deleted after 90 days. 
So for me: 
If something lives in /tmp folder longer than 90 days and still needed would be definitive at wrong folder. 

The only thing we could think about it would be to have a minimum number of days to wait e.g. 14 days or something

Reply:
@zoidbergx 
> If something lives in /tmp folder longer than 90 days and still needed would be definitive at wrong folder.

Not really, might be a question of definition, but in Pimcore's case, the /tmp folder is cleaned up implicitly when the dependent source file changes. In case of thumbnails, this means that the image thumbnail files will stay on /tmp as long as the original file is unchanged and also the thumbnail definition doesn't change. Imho it doesn't make sense to re-generate thumbnail files just for the sake of it 😊